resources/sts/4.14/hypershift/sts_extended_hcp_support_permission_policy.json

```diff
@@ -1,115 +1,370 @@
-                "ec2:DescribeVpcAttribute",
-                "ec2:DescribeVpcClassicLink",
-                "ec2:DescribeVpcClassicLinkDnsSupport",
-                "ec2:DescribeVpcEndpointConnectionNotifications",
-                "ec2:DescribeVpcEndpointServiceConfigurations",
-                "ec2:DescribeVpcEndpointServicePermissions",
-                "ec2:DescribeVpcEndpointServices",
-                "ec2:DescribeVpcPeeringConnections",
-                "ec2:DescribeVpnConnections",
-                "ec2:DescribeVpnGateways",
-                "ec2:GetAssociatedIpv6PoolCidrs",
-                "ec2:GetManagedPrefixListEntries",
-                "ec2:GetSerialConsoleAccessStatus",
-                "ec2:GetTransitGatewayAttachmentPropagations",
-                "ec2:GetTransitGatewayMulticastDomainAssociations",
-                "ec2:GetTransitGatewayPrefixListReferences",
-                "ec2:GetTransitGatewayRouteTableAssociations",
-                "ec2:GetTransitGatewayRouteTablePropagations",
-                "ec2:ModifyInstanceAttribute",
-                "ec2:SearchLocalGatewayRoutes",
-                "ec2:SearchTransitGatewayMulticastGroups",
-                "ec2:SearchTransitGatewayRoutes",
-                "ec2:StartNetworkInsightsAnalysis",
-                "elasticloadbalancing:ConfigureHealthCheck",
-                "kms:CreateGrant",
-                "s3:GetBucketTagging",
-                "s3:GetObjectAcl",
-                "s3:GetObjectTagging",
-                "s3:ListAllMyBuckets",
-                "tiros:CreateQuery",
-                "tiros:GetQueryAnswer",
-                "tiros:GetQueryExplanation",
-                "ec2:RunInstances",
-                "ec2:DescribeInstanceTypes",
-                "ec2:CreateTags",
-                "ec2:GetConsoleOutput"
-            ],
-            "Resource": "*"
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Sid": "CreateSnapShotsRequest",
+      "Effect": "Allow",
+      "Action": [
+        "ec2:CreateSnapshots"
+      ],
+      "Resource": "arn:aws:ec2:*:*:snapshot/*",
+      "Condition": {
+        "StringEquals": {
+          "aws:RequestTag/red-hat-managed": "true"
+        }
+      }
+    },
+    {
+      "Sid": "CreateSnapShotsOtherResources",
+      "Effect": "Allow",
+      "Action": [
+        "ec2:CreateSnapshots"
+      ],
+      "Resource": [
+        "arn:aws:ec2:*:*:instance/*",
+        "arn:aws:ec2:*:*:volume/*"
+      ]
+    },
+    {
+      "Sid": "DescribeSnapShotAttribute",
+      "Effect": "Allow",
+      "Action": [
+        "ec2:DescribeSnapshotAttribute"
+      ],
+      "Resource": "arn:aws:ec2:*:*:snapshot/*",
+      "Condition": {
+        "StringEquals": {
+          "aws:ResourceTag/red-hat-managed": "true"
+        }
+      }
+    },
+    {
+      "Sid": "DescribeSnapShot",
+      "Effect": "Allow",
+      "Action": [
+        "ec2:DescribeSnapshots"
+      ],
+      "Resource": "*"
```

Comment:
Since all snapshots are tagged, we can condition this

Reply:
I have tested this, and it it can't seem to be conditioned, and the chart [here](https://docs.aws.amazon.com/service-authorization/latest/reference/list_amazonec2.html#amazonec2-internet-gateway) it does not take aws:ResourceTag/${TagKey}

Reply:
Thanks!